services/functional-tests/src/test/java/org/infinispan/online/service/datagrid/PersistedDataSurvivesTest.java

```diff
@@ -0,0 +1,107 @@
+package org.infinispan.online.service.datagrid;
+
+import io.fabric8.openshift.client.OpenShiftClient;
+import org.arquillian.cube.openshift.impl.requirement.RequiresOpenshift;
+import org.arquillian.cube.requirement.ArquillianConditionalRunner;
+import org.infinispan.commons.configuration.XMLStringConfiguration;
+import org.infinispan.online.service.endpoint.HotRodTester;
+import org.infinispan.online.service.scaling.ScalingTester;
+import org.infinispan.online.service.utils.DeploymentHelper;
+import org.infinispan.online.service.utils.OpenShiftClientCreator;
+import org.infinispan.online.service.utils.OpenShiftCommandlineClient;
+import org.infinispan.online.service.utils.OpenShiftHandle;
+import org.infinispan.online.service.utils.ReadinessCheck;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.InSequence;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+@RunWith(ArquillianConditionalRunner.class)
+@RequiresOpenshift
+public class PersistedDataSurvivesTest {
+
+   private static final String SERVICE_NAME = "datagrid-service";
+   private static final String CACHE_NAME = "custom-persistent";
+
+   private OpenShiftClient client = OpenShiftClientCreator.getClient();
+
+   private ReadinessCheck readinessCheck = new ReadinessCheck();
+   private OpenShiftHandle handle = new OpenShiftHandle(client);
+
+   private ScalingTester scalingTester = new ScalingTester();
+   private OpenShiftCommandlineClient commandlineClient = new OpenShiftCommandlineClient();
+
+   @Deployment
+   public static Archive<?> deploymentApp() {
+      return ShrinkWrap
+         .create(WebArchive.class, "test.war")
+         .addAsLibraries(DeploymentHelper.testLibs())
+         .addPackage(ReadinessCheck.class.getPackage())
+         .addPackage(ScalingTester.class.getPackage())
+         .addPackage(HotRodTester.class.getPackage());
+   }
+
+   @Before
+   public void before() {
+      readinessCheck.waitUntilAllPodsAreReady(client);
+   }
+
+   @InSequence(1)
+   @Test
+   public void put_on_persisted_cache() throws Exception {
+      URL hotRodService = handle.getServiceWithName(SERVICE_NAME + "-hotrod");
+      HotRodTester hotRodTester = new HotRodTester(SERVICE_NAME, hotRodService, client);
+
+      String cacheCfg = String.format(
+         "<infinispan>" +
+            "<cache-container>" +
+               "<distributed-cache name=\"%1$s\">" +
```

Comment:
@galderz Does the user always have to provide the `<infinispan><cache-container>` elements? It seems redundant to me as I assume that the created configurations will always be apart of the same cache-container.